### DISCUSSION TITLE
Add Deno as a default app

### DISCUSSION THREAD

**Discussion by Himadu2000** [nrwl/nx#5823] [Feature Requests]
Deno is currently gaining lot of support and the de-facto backend for many of apps, so It is a must to add deno as a default app in nx workspace.

**Comment by algoflows** [comment]
Any movement on this, is this planned?

**Comment by SimonVillage** [comment]
Is someone willing to write a simple package to at least generate the application and to build / serve it?

**Reply by andyjessop** [reply]
This sounds like a fun little project. Are we talking about Fresh here? That seems like the most popular Deno framework at the moment. Another project would be to build generators for libs that use Deno to lint/test/build.

**Reply by ggwzrd** [reply]
It seems that someone already attempted something https://github.com/codemonument/nx-deno

**Comment by shahoob** [comment]


**Reply by dnim** [reply]
why not.

**Comment by barbados-clemens** [comment]
Hi all, we are working on `@nrwl/deno` in the nx-labs repo right now. 
initial goals:
1. be able to build an app/edge function in deno
3. nx project graph support

Here is a little demo showing
1. A sample app and lib being generated
2. using the lib in the app using an import alias

https://user-images.githubusercontent.com/23272162/217955412-2ba9e40e-463c-483c-bbac-31d330ed6c57.mp4


there are 
1. app/lib generators
2. build/serve/test/lint executors
3. project graph support

in the PR right and I'm hoping to get a beta cut soon for it. 
https://github.com/nrwl/nx-labs/pull/138

**Comment by b2m-marcjovani** [comment]
That sounds great @barbados-clemens! Here's someone wanting to try it out on their Nest + Angular monorepo. :smiley:

**Reply by barbados-clemens** [reply]
Always open to feedback. 

The plugin has been around for a bit and there have been some changes since initial publish mainly around 'building' in deno.

install `@nx/deno` and giving it a try!

**Comment by drackp2m** [comment]
> Always open to feedback.
> 
> The plugin has been around for a bit and there have been some changes since initial publish mainly around 'building' in deno.
> 
> install `@nx/deno` and giving it a try!

Good barbados!!

Is there any public project that you have done in nx-labs, that I can take as inspiration to try to get it to run in my project that has been using `@nx/webpack:webpack` / `@angular-devkit/build-angular:browser` for a long time?

Thank you so much.

**Reply by barbados-clemens** [reply]
Most likely cannot just switch out with webpack or angular builders as there will be all kinds of other issues you'll run into. mainly bc those are build steps and deno doesn't necessarily need one, though you can if you wish. 

here is an sample api app that might be helpful. 
https://github.com/nrwl/nx-recipes/tree/main/deno-deploy

**Comment by guillempuche** [comment]
The `@nx/deno` is deprecated https://github.com/nrwl/nx-labs/issues/400.

Could the team maintain it again?

**Comment by thenglong** [comment]
Deno 2 is out and it's very compatible with node. Is there a plan to re-add Deno as the default support?

**Reply by muhamedkarajic** [reply]
Add deno 2 so we finally can decently unit test functions.

**Comment by FredrikZeiner** [comment]
> Deno 2 is out and it's very compatible with node. Is there a plan to re-add Deno as the default support? 

Please do!

**Comment by timseriakov** [comment]
> Deno 2 is out and it's very compatible with node. Is there a plan to re-add Deno as the default support?

please please please

**Comment by RafaRochaS91** [comment]
Anything here? Or any link to an issue that can be worked on?
This would be a game changer to many companies specially in the world of LLM agents - the compilation potential plus backwards compatibility with npm makes deno the reason to some companies I know including my own to be using smth like turbo instead even though Nx is far more mature in every aspect.
Thanks for all the work though and let us know how could we help!

**Comment by muhamedkarajic** [comment]
Is it a ridiculous idea to have a look at this? I see still nothing have been said after 1 year.
